modules/gcloud/src/main/scala/com/dimafeng/testcontainers/BigtableEmulatorContainer.scala

```diff
@@ -0,0 +1,65 @@
+package com.dimafeng.testcontainers
+
+import com.google.cloud.bigtable.admin.v2.{
+  BigtableTableAdminClient,
+  BigtableTableAdminSettings
+}
+import com.google.cloud.bigtable.data.v2.{
+  BigtableDataClient,
+  BigtableDataSettings
+}
+import org.testcontainers.containers.{
+  BigtableEmulatorContainer => JavaBigtableEmulatorContainer
+}
+import org.testcontainers.utility.DockerImageName
+
+class BigtableEmulatorContainer(
+  bigtableEmulatorImageName: Option[DockerImageName] = None,
+  projectId: String = BigtableEmulatorContainer.defaultProjectId,
+  instanceId: String = BigtableEmulatorContainer.defaultInstanceId
+) extends SingleContainer[JavaBigtableEmulatorContainer] {
+
+  override val container: JavaBigtableEmulatorContainer =
+    bigtableEmulatorImageName
+      .map(new JavaBigtableEmulatorContainer(_))
+      .getOrElse(
+        new JavaBigtableEmulatorContainer(
+          BigtableEmulatorContainer.defaultImageName
+        )
+      )
+
+  lazy val dataClient: BigtableDataClient =
+    BigtableDataClient.create(
+      BigtableDataSettings
+        .newBuilderForEmulator(container.getHost, container.getEmulatorPort)
+        .setProjectId(projectId)
+        .setInstanceId(instanceId)
+        .build()
+    )
+
+  lazy val tableAdminClient: BigtableTableAdminClient =
+    BigtableTableAdminClient.create(
+      BigtableTableAdminSettings
+        .newBuilderForEmulator(container.getHost, container.getEmulatorPort)
+        .setProjectId(projectId)
+        .setInstanceId(instanceId)
+        .build()
+    )
+
+  def emulatorHost: String = container.getHost
+  def emulatorPort: Int = container.getEmulatorPort
+}
+
+object BigtableEmulatorContainer {
+
+  val defaultProjectId = "test-project"
+  val defaultInstanceId = "test-instance"
+
+  val defaultImageName: DockerImageName =
+    DockerImageName.parse("gcr.io/google.com/cloudsdktool/cloud-sdk")
+
+  def apply(
+    bigtableEmulatorImageName: DockerImageName = null
```

Comment:
would it make sense to add `projectId` and `instanceId` to this method as well?

Reply:
If we do that, I'm not sure it's adding anything over the standard class constructor where all of the 3 parameters have default values anyway: 

```scala
class BigtableEmulatorContainer(
  bigtableEmulatorImageName: Option[DockerImageName] = None,
  projectId: String = BigtableEmulatorContainer.defaultProjectId,
  instanceId: String = BigtableEmulatorContainer.defaultInstanceId
)
```

So it might make more sense to just drop the `apply` method (or keep it as something that offers a different signature), but I'll go with whatever you prefer here.

Reply:
I think it's fine, that's the pattern we established initially. Or you can drop the `apply` and make the class a case class

Reply:
OK, have gone with the case class approach 👍